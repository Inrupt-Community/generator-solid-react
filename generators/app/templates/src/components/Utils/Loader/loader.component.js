import React, { Component } from "react";
import { LoaderWrapper, CubeGrid, Cube } from "./loader.style";
type Props = {
  show: boolean
};

class Loader extends Component<Props> {
  render() {
    const { show } = this.props;
    const cubes = [0.2, 0.3, 0.4, 0.1, 0.2, 0.3, 0, 0.1, 0.2];

    return (
      <LoaderWrapper show={show}>
        <CubeGrid>
          {cubes.map((delay, i) => (
            <Cube key={i} delay={delay} />
          ))}
        </CubeGrid>
      </LoaderWrapper>
    );
  }
}

export default Loader;
