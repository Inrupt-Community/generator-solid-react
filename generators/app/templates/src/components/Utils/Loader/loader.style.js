import styled, { keyframes } from "styled-components";

export const LoaderWrapper = styled.div`
  position: fixed;
  z-index: 9999;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  align-items: center;
  display: ${({ show }) => (show ? "flex" : "none")};
  background-color: rgba(255, 255, 255, 0.6);

  .sk-cube-grid {
    width: 80px;
    height: 80px;
    margin: 100px auto;

    .sk-cube {
      width: calc(100% / 3);
      height: calc(100% / 3);
      background-color: $purple;
      float: left;
      animation: ${cubeGridScaleDelay} 1.3s infinite ease-in-out;
    }

    .sk-cube1 {
      animation-delay: 0.2s;
    }

    .sk-cube2 {
      animation-delay: 0.3s;
    }

    .sk-cube3 {
      animation-delay: 0.4s;
    }

    .sk-cube4 {
      animation-delay: 0.1s;
    }

    .sk-cube5 {
      animation-delay: 0.2s;
    }

    .sk-cube6 {
      animation-delay: 0.3s;
    }

    .sk-cube7 {
      animation-delay: 0s;
    }

    .sk-cube8 {
      animation-delay: 0.1s;
    }

    .sk-cube9 {
      animation-delay: 0.2s;
    }
  }
`;

const cubeGridScaleDelay = keyframes`
	0%,
	70%,
	100% {
		transform: scale3D(1, 1, 1);
	}

	35% {
		transform: scale3D(0, 0, 1);
	}
`;
