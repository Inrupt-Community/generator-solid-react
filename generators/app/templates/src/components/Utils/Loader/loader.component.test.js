import React from 'react';
import Loader from './';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
	const component = shallow(<Loader />);

	expect(component).toMatchSnapshot();
});
