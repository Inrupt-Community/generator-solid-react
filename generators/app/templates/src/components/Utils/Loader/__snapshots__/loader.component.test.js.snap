// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="sk-cube-grid"
      >
        <div
          className="sk-cube sk-cube1"
        />
        <div
          className="sk-cube sk-cube2"
        />
        <div
          className="sk-cube sk-cube3"
        />
        <div
          className="sk-cube sk-cube4"
        />
        <div
          className="sk-cube sk-cube5"
        />
        <div
          className="sk-cube sk-cube6"
        />
        <div
          className="sk-cube sk-cube7"
        />
        <div
          className="sk-cube sk-cube8"
        />
        <div
          className="sk-cube sk-cube9"
        />
      </div>,
      "show": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sk-cube sk-cube1"
          />,
          <div
            className="sk-cube sk-cube2"
          />,
          <div
            className="sk-cube sk-cube3"
          />,
          <div
            className="sk-cube sk-cube4"
          />,
          <div
            className="sk-cube sk-cube5"
          />,
          <div
            className="sk-cube sk-cube6"
          />,
          <div
            className="sk-cube sk-cube7"
          />,
          <div
            className="sk-cube sk-cube8"
          />,
          <div
            className="sk-cube sk-cube9"
          />,
        ],
        "className": "sk-cube-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube3",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube4",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube5",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube6",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube7",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube8",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sk-cube sk-cube9",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  position: fixed;
  z-index: 9999;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  align-items: center;
  display: ",
          [Function],
          ";
  background-color: rgba(255, 255, 255, 0.6);

  .sk-cube-grid {
    width: 80px;
    height: 80px;
    margin: 100px auto;

    .sk-cube {
      width: calc(100% / 3);
      height: calc(100% / 3);
      background-color: $purple;
      float: left;
      animation: ",
          Keyframes {
            "id": "sc-keyframes-hLIxZe",
            "inject": [Function],
            "name": "hLIxZe",
            "rules": Array [
              "@-webkit-keyframes hLIxZe{0%,70%,100%{-webkit-transform:scale3D(1,1,1);-ms-transform:scale3D(1,1,1);transform:scale3D(1,1,1);}35%{-webkit-transform:scale3D(0,0,1);-ms-transform:scale3D(0,0,1);transform:scale3D(0,0,1);}}",
              "@keyframes hLIxZe{0%,70%,100%{-webkit-transform:scale3D(1,1,1);-ms-transform:scale3D(1,1,1);transform:scale3D(1,1,1);}35%{-webkit-transform:scale3D(0,0,1);-ms-transform:scale3D(0,0,1);transform:scale3D(0,0,1);}}",
            ],
            "toString": [Function],
          },
          " 1.3s infinite ease-in-out;
    }

    .sk-cube1 {
      animation-delay: 0.2s;
    }

    .sk-cube2 {
      animation-delay: 0.3s;
    }

    .sk-cube3 {
      animation-delay: 0.4s;
    }

    .sk-cube4 {
      animation-delay: 0.1s;
    }

    .sk-cube5 {
      animation-delay: 0.2s;
    }

    .sk-cube6 {
      animation-delay: 0.3s;
    }

    .sk-cube7 {
      animation-delay: 0s;
    }

    .sk-cube8 {
      animation-delay: 0.1s;
    }

    .sk-cube9 {
      animation-delay: 0.2s;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="sk-cube-grid"
        >
          <div
            className="sk-cube sk-cube1"
          />
          <div
            className="sk-cube sk-cube2"
          />
          <div
            className="sk-cube sk-cube3"
          />
          <div
            className="sk-cube sk-cube4"
          />
          <div
            className="sk-cube sk-cube5"
          />
          <div
            className="sk-cube sk-cube6"
          />
          <div
            className="sk-cube sk-cube7"
          />
          <div
            className="sk-cube sk-cube8"
          />
          <div
            className="sk-cube sk-cube9"
          />
        </div>,
        "show": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="sk-cube sk-cube1"
            />,
            <div
              className="sk-cube sk-cube2"
            />,
            <div
              className="sk-cube sk-cube3"
            />,
            <div
              className="sk-cube sk-cube4"
            />,
            <div
              className="sk-cube sk-cube5"
            />,
            <div
              className="sk-cube sk-cube6"
            />,
            <div
              className="sk-cube sk-cube7"
            />,
            <div
              className="sk-cube sk-cube8"
            />,
            <div
              className="sk-cube sk-cube9"
            />,
          ],
          "className": "sk-cube-grid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube1",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube4",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube5",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube6",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube7",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube8",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sk-cube sk-cube9",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  position: fixed;
  z-index: 9999;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  align-items: center;
  display: ",
            [Function],
            ";
  background-color: rgba(255, 255, 255, 0.6);

  .sk-cube-grid {
    width: 80px;
    height: 80px;
    margin: 100px auto;

    .sk-cube {
      width: calc(100% / 3);
      height: calc(100% / 3);
      background-color: $purple;
      float: left;
      animation: ",
            Keyframes {
              "id": "sc-keyframes-hLIxZe",
              "inject": [Function],
              "name": "hLIxZe",
              "rules": Array [
                "@-webkit-keyframes hLIxZe{0%,70%,100%{-webkit-transform:scale3D(1,1,1);-ms-transform:scale3D(1,1,1);transform:scale3D(1,1,1);}35%{-webkit-transform:scale3D(0,0,1);-ms-transform:scale3D(0,0,1);transform:scale3D(0,0,1);}}",
                "@keyframes hLIxZe{0%,70%,100%{-webkit-transform:scale3D(1,1,1);-ms-transform:scale3D(1,1,1);transform:scale3D(1,1,1);}35%{-webkit-transform:scale3D(0,0,1);-ms-transform:scale3D(0,0,1);transform:scale3D(0,0,1);}}",
              ],
              "toString": [Function],
            },
            " 1.3s infinite ease-in-out;
    }

    .sk-cube1 {
      animation-delay: 0.2s;
    }

    .sk-cube2 {
      animation-delay: 0.3s;
    }

    .sk-cube3 {
      animation-delay: 0.4s;
    }

    .sk-cube4 {
      animation-delay: 0.1s;
    }

    .sk-cube5 {
      animation-delay: 0.2s;
    }

    .sk-cube6 {
      animation-delay: 0.3s;
    }

    .sk-cube7 {
      animation-delay: 0s;
    }

    .sk-cube8 {
      animation-delay: 0.1s;
    }

    .sk-cube9 {
      animation-delay: 0.2s;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
